modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientComputeTest.java

```diff
@@ -321,63 +315,164 @@ void testExceptionInJobPropagatesToClientWithClassAndMessage(boolean async) {
 
         if (async) {
             JobExecution<String> execution = client().compute().executeAsync(Set.of(node(0)), List.of(), ExceptionJob.class.getName());
-
-            CompletionException ex = assertThrows(
-                    CompletionException.class,
-                    () -> execution.resultAsync().join()
+            cause = getExceptionInJobExecutionAsync(execution);
+        } else {
+            cause = getExceptionInJobExecutionSync(
+                    () -> client().compute().execute(Set.of(node(0)), List.of(), ExceptionJob.class.getName())
             );
+        }
 
-            assertThat(execution.statusAsync(), willBe(jobStatusWithState(FAILED)));
+        assertComputeExceptionWithClassAndMessage(cause);
+    }
 
-            cause = (IgniteException) ex.getCause();
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
```

Comment:
I think we can drop this parametrized thing and inline the logic into each test method.